### PR TITLE
Part of #1145: Added setup for frontend testing and Added test for Spinner component

### PR DESCRIPTION
**Pull request by tonyvugithub**
## Issue This PR Addresses
Part of #1145. What's in this PR?
1. Set up files required for testing
2. Provided tests for Spinner component
3. Changed jest config in package.json to ignore frontend test when run `npm run test`

## Type of Change
- [x] Add frontend tests

## Description
I would like to provide details on what I did to set up testing environment for front end
1. I followed Gatsby documentation on [Unit Testing](https://www.gatsbyjs.com/docs/unit-testing/) and [Testing React Components](https://www.gatsbyjs.com/docs/testing-react-components/). And those extra files in the root folders as well as the `__mock__` folder are the result of it. 

2. In my test, I added the `data-testid` attribute to the component so I can use it to refer to specific part of the component for testing. And because of that I included the custom `.babelrc` file to customize the default babel from Gatsby. What this file does is to remove the `data-testid` attribute I have added to the component so It won't show on browsers both in dev and prod. To do this I need two packages to be install including *babel-preset-gatsby* (to customize babel in Gatsby) and *babel-plugin-react-remove-properties* (to remove `data-testid` attribute before rendering)

3. I also provided tests for Spinner component as sample test case for you to have an idea how frontend testing works. You can find the file `Spinner.test.js` under `frontend/test` folder

4. Since jest will attempt to cover the front end tests  if we run `npm run test` in root. This would eventually lead to failures in backend test setup. I have added extra ignorePattern for jest in package.json to ignore frontend tests when run tests for backend. Also, I have added a `test-frontend` script in `package.json` to run it separately with the main set up. Now if you run `npm run test` jest will first run just the backend tests then run `test-frontend` to run frontend tests. By doing so, we would have no conflict in set up and we can have the results for both tests in one command.   

## Checklist

- [ ] **Quality**: This PR builds and passes our npm test and works locally
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not (if applicable)
- [ ] **Documentation**: This PR includes updated/added documentation to user exposed functionality or configuration variables are added/changed or an explanation of why it does not(if applicable)

### CODE REVIEW

**Comment by tonyvugithub** [comment]
@humphd : I think it failed ci because the backend test run my front end tests as well.

File: .vs/ProjectSettings.json
Comment (raygervais) [inline .vs/ProjectSettings.json:1]: Can we ignore adding .vs (Visual Studio, VSCode) files to the project? Maybe it's just my own nitpick but unsure if it adds any value.

**Review by raygervais — CHANGES_REQUESTED** [review]

File: .vs/ProjectSettings.json
Comment (tonyvugithub) [reply .vs/ProjectSettings.json:1]: my bad. I was being lazy and use add . let me get rid of it

Comment (tonyvugithub) [reply .vs/ProjectSettings.json]: @raygervais : new commit pushed without .vs files

**Review by manekenpix — APPROVED** [review]

**Review by cindyledev — APPROVED** [review]

**Comment by tonyvugithub** [comment]
@raygervais  @c3ho : Can someone take a last look at this? This has been waiting to merge for a while now

**Comment by raygervais** [comment]
@tonyvugithub, you're good to merge if you have the other two approvals. :)

Word of warning, when you add multiple reviewers you now add more
blocking points in the way of a merge.

**Review by raygervais — APPROVED** [review]

**Comment by tonyvugithub** [comment]
@raygervais : I think Josue might have added more maintainers to it. Thank you for the warning, I keep it in mind for future PR

**Comment by manekenpix** [comment]
The problem was that, since @raygervais requested changes, github wouldn't let us merge the PR until @raygervais approved it (the PR had 2 approvals but it was still blocked).

**Comment by tonyvugithub** [comment]
@manekenpix : That makes sense! Anyway, thanks guys for all the feedbacks!

**Pull request merged**
